server/src/main/java/org/elasticsearch/search/aggregations/metrics/GlobalOrdCardinalityAggregator.java

```diff
@@ -0,0 +1,203 @@
+
+    private final ValuesSource.Bytes.WithOrdinals valuesSource;
+
+    // Build at post-collection phase
+    @Nullable
+    private HyperLogLogPlusPlusSparse counts;
+
+    private OrdinalsCollector collector;
+
+    private final int precision;
+
+    public GlobalOrdCardinalityAggregator(
+            String name,
+            ValuesSource.Bytes.WithOrdinals valuesSource,
+            int precision,
+            int maxOrd,
+            SearchContext context,
+            Aggregator parent,
+            Map<String, Object> metadata) throws IOException {
+        super(name, context, parent, metadata);
+        this.valuesSource = valuesSource;
+        this.counts = null;
+        this.collector = new OrdinalsCollector(context.bigArrays(), maxOrd);
+        this.precision = precision;
+    }
+
+    @Override
+    public ScoreMode scoreMode() {
+        return valuesSource.needsScores() ? ScoreMode.COMPLETE : ScoreMode.COMPLETE_NO_SCORES;
+    }
+
+    @Override
+    public LeafBucketCollector getLeafCollector(LeafReaderContext ctx,
+            final LeafBucketCollector sub) throws IOException {
+        this.collector.set(valuesSource.globalOrdinalsValues(ctx));
+        return collector;
+    }
+
+    @Override
+    protected void doPostCollection() throws IOException {
+        counts = new HyperLogLogPlusPlusSparse(precision, collector.bigArrays, collector.visitedOrds.size());
+        collector.postCollect(counts);
+        collector.close();
+        collector = null;
+    }
+
+    @Override
+    public double metric(long owningBucketOrd) {
+        return counts == null ? 0 : counts.cardinality(owningBucketOrd);
+    }
+
+    @Override
+    public InternalAggregation buildAggregation(long owningBucketOrdinal) {
+        if (counts == null || owningBucketOrdinal >= counts.maxOrd() || counts.cardinality(owningBucketOrdinal) == 0) {
+            return buildEmptyAggregation();
+        }
+        // We need to build a copy because the returned Aggregation needs remain usable after
+        // this Aggregator (and its HLL++ counters) is released.
+        AbstractHyperLogLogPlusPlus copy = counts.clone(owningBucketOrdinal, BigArrays.NON_RECYCLING_INSTANCE);
+        return new InternalCardinality(name, copy, metadata());
+    }
+
+    @Override
+    public InternalAggregation buildEmptyAggregation() {
+        return new InternalCardinality(name, null, metadata());
+    }
+
+    @Override
+    protected void doClose() {
+        Releasables.close(counts, collector);
+    }
+
+    @Override
+    public void collectDebugInfo(BiConsumer<String, Object> add) {
+        // maybe need to add something for global?
+        super.collectDebugInfo(add);
+        add.accept("empty_collectors_used", 0);
+        add.accept("numeric_collectors_used", 0);
+        add.accept("ordinals_collectors_used", 1);
+        add.accept("ordinals_collectors_overhead_too_high", 0);
+        add.accept("string_hashing_collectors_used", 0);
```

Comment:
We don't need these at all. We'll get the information that this is the global ords-based aggregator by default in the debug info. If we want anything here is might be debug insight into the sparse hll++ data structure.

Reply:
I think I saw some errors if I don't do this. I will double check.

Reply:
That probably comes from a test I just added. We can totally change it.

Reply:
I changed the test and remove it.